### PR TITLE
Sodium deprecations

### PR DESCRIPTION
**Pull request by waynew**
### What does this PR do?

Removes all the places that were marked with 'warn_until` and Sodium

### What issues does this PR fix or reference?

Several - all referenced in the commit messages

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

File: salt/modules/napalm_network.py [hunk 1]
```diff
@@ -1982,158 +1966,112 @@ def load_template(
...
-                template_hash_name = [None] * len(template_name)
-            if (
-                template_hash
-                and isinstance(template_hash, six.string_types)
-                and not (
-                    template_hash.startswith("salt://")
-                    or template_hash.startswith("file://")
-                )
+            if not isinstance(_managed, (list, tuple)) and isinstance(
+                _managed, six.string_types
```
Comment (s0undt3ch) [inline salt/modules/napalm_network.py:2048]: We could take the time and stop using `six.string_types` in favor or `str` on places where code is updated.

**Review by s0undt3ch — DISMISSED** [review]

**Review by DmitryKuzmenko — DISMISSED** [review]

**Comment by DmitryKuzmenko** [comment]
The tests failures are related BTW.

File: doc/ref/modules/all/salt.modules.keystone.rst [hunk 2]
```diff
@@ -1,6 +0,0 @@
-=====================
```
Comment (DmitryKuzmenko) [inline doc/ref/modules/all/salt.modules.keystone.rst:1]: As I can see the keystone module was deprecated and have to be removed but I don't see removal of the module in this PR.

Comment (waynew) [reply doc/ref/modules/all/salt.modules.keystone.rst:1]: That's been punted to the next release - the state was not deprecated, and hasn't been re-written to take advantage of the `keystoneng` module. Looks like I need to put the docs back, though :stuck_out_tongue_closed_eyes:

**Comment by waynew** [comment]
re-run full all

**Comment by dwoz** [comment]
re-run ubuntu

**Comment by max-arnold** [comment]
The old `module.run` syntax deprecation note was still pointing to 3001. Fixed in https://github.com/saltstack/salt/pull/57914/commits/549461921071e168269d7edc6093e33b81e453d8

**Pull request merged**